### PR TITLE
automatic gpu selection no longer works in CUDA-TK 7.0

### PR DESCRIPTION
**Pull request by vesis84**
- fixing an issue introduced with CUDA-TK 7.0, when automatic selection
  of free gpu by os / library no longer works. Library always selects '0',
  which leads to crash in case of 'exclusive mode' when '0' is taken and 
  other gpus are free.
- this fix is inspired by Dan Povey's fix for Kaldi:
  kaldi-asr/kaldi@6548565
- it does a loop over gpus until a free gpu is taken.
- pycuda initialization was also affected by the bug from CUDA 7.0, so it is better to disable it.

### CODE REVIEW

File: theano/sandbox/cuda/__init__.py [hunk 1]
```diff
@@ -399,9 +399,9 @@ def use(device,
 
         # Has PyCUDA already initialized the GPU context
         pycuda_init_dev = False
-        if config.pycuda.init:
-            import theano.misc.pycuda_init
-            pycuda_init_dev = theano.misc.pycuda_init.pycuda_available
+#        if config.pycuda.init:
```
Comment (nouiz) [inline theano/sandbox/cuda/__init__.py]: Why this? By default pycuda.init is False, so by default it do nothing.

**Comment by nouiz** [comment]
This all seem good. I tested it and we also had this problem, but never investigated. Thanks!

**Comment by nouiz** [comment]
When travis pass, we can merge.

**Comment by nouiz** [comment]
I forgot, what about the pycuda change? I don't understand why this is needed.

**Comment by vesis84** [comment]
The pycuda change is not needed as long as the pycuda gpu selection is disabled by default. The problem will reappear only in the case if someone enables pycuda gpu selection and has CUDA 7.0. So there could be some warning message instead of disabling pycuda gpu selection?

**Comment by vesis84** [comment]
Other reason to disable 'pycuda' is that we have seen a situation when 2GPUs were taken by theano. One was taken by pycuda and the other with our modified cuda_ndarray.cuda_ndarray.select_a_gpu().
It disappeard after commenting the pycuda initialization. Is pycuda something that many people use and depend on?

**Comment by vesis84** [comment]
Hi, I changed it a bit. Now if the gpu is selected by pycuda, we don't call cuda_ndarray.select_a_gpu() anymore. While there is the initialization for cublas, etc... Cheers! K.

**Comment by nouiz** [comment]
that seem the right fix. We can merge when travis pass.

thanks

**Pull request merged**